src/common/helpers/data-extract/tp-application.test.js

```diff
@@ -0,0 +1,329 @@
+
+    it('should return empty string for unmatched combinations', () => {
+      mockApplicationData.sections
+        .at(1)
+        .questionAnswers.find(
+          (qa) => qa.questionKey === 'originType'
+        ).answer.value = 'market'
+      mockApplicationData.sections
+        .at(2)
+        .questionAnswers.find(
+          (qa) => qa.questionKey === 'destinationType'
+        ).answer.value = 'market'
+      const application = new TbApplication(mockApplicationData)
+
+      expect(application.licenceType).toBe('')
+    })
+  })
+
+  describe('requesterCphNumber', () => {
+    it('should return destination CPH when on farm', () => {
+      mockApplicationData.sections
+        .at(1)
+        .questionAnswers.find(
+          (qa) => qa.questionKey === 'onOffFarm'
+        ).answer.value = 'on'
+      const application = new TbApplication(mockApplicationData)
+
+      expect(application.requesterCphNumber).toBe('98/765/4321')
+    })
+
+    it('should return origin CPH when off farm', () => {
+      const application = new TbApplication(mockApplicationData)
+
+      expect(application.requesterCphNumber).toBe('12/345/6789')
+    })
+
+    it('should return empty string when origin section does not exist', () => {
+      mockApplicationData.sections = mockApplicationData.sections.filter(
+        (s) => s.sectionKey !== 'origin'
+      )
+      const application = new TbApplication(mockApplicationData)
+
+      expect(application.requesterCphNumber).toBe('')
+    })
+
+    it('should return empty string when destination section does not exist', () => {
+      mockApplicationData.sections
+        .at(1)
+        .questionAnswers.find(
+          (qa) => qa.questionKey === 'onOffFarm'
+        ).answer.value = 'on'
+      mockApplicationData.sections = mockApplicationData.sections.filter(
+        (s) => s.sectionKey !== 'destination'
+      )
+      const application = new TbApplication(mockApplicationData)
+
+      expect(application.requesterCphNumber).toBe('')
+    })
+
+    it('should return empty string when CPH questions do not exist', () => {
+      mockApplicationData.sections.at(1).questionAnswers =
+        mockApplicationData.sections
+          .at(1)
+          .questionAnswers.filter((qa) => qa.questionKey !== 'cphNumber')
+      mockApplicationData.sections.at(2).questionAnswers =
+        mockApplicationData.sections
+          .at(2)
+          .questionAnswers.filter(
+            (qa) => qa.questionKey !== 'destinationFarmCph'
+          )
+      const application = new TbApplication(mockApplicationData)
+
+      expect(application.requesterCphNumber).toBe('')
+    })
+  })
+
+  describe('inheritance', () => {
+    it('should inherit from Application class', () => {
+      const application = new TbApplication(mockApplicationData)
+
+      expect(application.journeyId).toBe(
+        'GET_PERMISSION_TO_MOVE_ANIMALS_UNDER_DISEASE_CONTROLS_TB_ENGLAND'
+      )
+      expect(typeof application.get).toBe('function')
```

Comment:
Same comment as above - a more direct inheritance check seems to align better to the stated purpose of the test